src/main/java/uk/gov/hmcts/dts/fact/services/validation/ValidationService.java

```diff
@@ -54,4 +56,23 @@ public List<String> validateFullPostcodes(List<String> postcodes) {
     public boolean validateLocalAuthority(final String localAuthorityName) {
         return mapitlocalAuthorityValidator.localAuthorityNameIsValid(localAuthorityName);
     }
+
+    /**
+     * Validates that the given epimId is alphanumeric and contains dashes only.
+     * @param courtAddresses A list of court addresses
+     * @return The epimId if it fails the regex, otherwise null
+     */
+    public String validateEpimIds(List<CourtAddress> courtAddresses) {
+        //alphanumeric and dashes only up to 30
+        String epimIdPattern = "^[a-zA-Z0-9-]{0,30}$";
```

Comment:
Could probably combine this into one line if you wanted to (this and below). Up to you though

Reply:
changed